### ISSUE TITLE
getScaleFromProps has inconsistent return values

### ISSUE THREAD

**Issue by kenwheeler** [FormidableLabs/victory-chart#341]
During a performance audit, I traced a big hit back to `getScaleTypeFromData`. In the course of investigation, I saw that `getScaleFromProps` in the context of `getScaleType` is returning inconsistent values. See a failing test case here:

https://github.com/FormidableLabs/victory-chart/compare/scale-value-perf-fix?expand=1

For example, in VictoryBar, there is a `getScaleType` check in `getBarPosition`, which was getting called for 2000 points. You can see that here: 

https://github.com/FormidableLabs/victory-chart/blob/master/src/components/victory-bar/helper-methods.js#L38

What happens is, when that call is made, and if you have explicitly supplied a string value in short or object form for `scale`, it returns a function from `getScaleFromProps` which then fails a string type check and fall back to `getScaleFromData` here: 

https://github.com/FormidableLabs/victory-chart/blob/master/src/helpers/scale.js#L72-L90

To recap the performance implications here, I profiled and react-addons-perf'd the before and after of making the statement in the second link evaluate properly:

**Before:** (1.8s render)

![screen shot 2016-08-13 at 1 12 08 am](https://cloud.githubusercontent.com/assets/286616/17641568/95d5eb44-60f3-11e6-89d4-70ba49be3ee4.png)

![screen shot 2016-08-13 at 1 12 40 am](https://cloud.githubusercontent.com/assets/286616/17641574/bcb86052-60f3-11e6-968a-a178023344ef.png)

**After:** (380ms Render)

![screen shot 2016-08-13 at 1 13 40 am](https://cloud.githubusercontent.com/assets/286616/17641571/9f1eb3b6-60f3-11e6-8fa5-d326956166dc.png)

![screen shot 2016-08-13 at 1 14 00 am](https://cloud.githubusercontent.com/assets/286616/17641572/a21ed29e-60f3-11e6-9e5c-87fc653d4461.png)
